### ISSUE TITLE
Unable to aquire BlobServiceClient instantiation with Azure Identity

### ISSUE THREAD

**Issue by SureshRangachariDXC** [Azure/azure-sdk-for-java#22940]
**Query/Question**
Unable to aquire BlobServiceClient instantiation with Azure Identity ending up with endless dependencies and getting an exception with below stack trace

javax.servlet.ServletException: com.fasterxml.jackson.dataformat.xml.XmlMapper.coercionConfigDefaults()Lcom/fasterxml/jackson/databind/cfg/MutableCoercionConfig;

@alzimmermsft -appreciate your quick response 

Full stack trace is attached with this thread.

***Why is this not a Bug or a feature Request?***

We feel that it could be that we are not using the right combination of Azure Blob Storage , Azure Identity with Jackson libraries. 

**Setup (please complete the following information if applicable):**
 - OS: Linux 
 - IDE:  Eclipse 2019-06
 - Library/Libraries: azure-storage-blob-12.12.0 , azure-identity 1.3.3 , jackson-core 2.12.4, jackson-annotations 2.12.4, jackson-module-jaxb-annotations 2.12.4, jackson-dataformat-xml 2.12.4, jackson-datatype-jsr310 2.12.4, jackson-databind 2.12.4
 - 
 
 **Information Checklist**
 Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [ x] Query Added
- [ x] Setup information Added
[server_log.txt](https://github.com/Azure/azure-sdk-for-java/files/6810149/server_log.txt)

**Comment by alzimmermsft** [comment]
Thank you for reporting this @SureshRangachariDXC, looking at Jackson 2.12.4 source code and the reflection logic being used in `JacksonAdapter` (the root call location in the SDKs before the exception triggers) the exception would be swallowed if the class or method didn't exist. Based on the line number of the exception is actually appears to be happening from outside of reflective usage, so it seems this is more likely a version conflict error. I know you listed some libraries being used to produce this issue but could you include a simplified POM and the `mvn dependency:tree` output so that I'm able to reproduce this locally.

**Comment by alzimmermsft** [comment]
cc: @lmolkova

**Comment by SureshRangachariDXC** [comment]
Hi Alan,

Firstly thank you for quick turn around on this one. Here you go with what you have requested from our end. Let me know for any further details

[simplified pom.txt](https://github.com/Azure/azure-sdk-for-java/files/6813480/simplified.pom.txt)
[mvn dependency-colon-tree.txt](https://github.com/Azure/azure-sdk-for-java/files/6813497/mvn.dependency-colon-tree.txt)

As you said could you please let us know the right combo of Azure Storage, Azure Identity, Azure Core, Jackson Lib Version.

Appreciate your help here.

Thank You

**Comment by SureshRangachariDXC** [comment]
mvn dependency tree with verbose
[mvn dependency tree - verbose.txt](https://github.com/Azure/azure-sdk-for-java/files/6816097/mvn.dependency.tree.-.verbose.txt)

**Comment by alzimmermsft** [comment]
Hi @SureshRangachariDXC, I attempted to recreate a simple application which only retrieves a blob's properties with the given dependencies but wasn't able to replicate the error. Is the application you're running hosted in another environment or is using a framework such as Spring?

**Comment by SureshRangachariDXC** [comment]
Hi Alan,
Yes !

Basically this is an JSF Primefaces based Project that gets deployed on JBoss Wildfly 19 run time. If you look at the maven dependency tree wildfly bom brings in all jackson 2.9 , netty related deps on its own. Is it something conflicting with azure blob storage dependencies. ?

Your thoughts

**Comment by SureshRangachariDXC** [comment]
@alzimmermsft @lmolkova 
Hi Alan,
Yes !

Basically this is an JSF Primefaces based Project that gets deployed on JBoss Wildfly 18 run time. If you look at the maven dependency tree wildfly bom brings in all jackson 2.9 , netty related deps on its own. Is it something conflicting with azure blob storage dependencies. ? So we are having issue with Dependency conflicts of wildfly 18 BOM and Azure Blobstorage Deps.

Note: For Azure Blob Storage by standalone Java application and i tired to do it as a shaded jar implementation the non azure dependency libraries that it packs with it like jackson, netty, ...etc might be conflicting with wildfly 18 run time depdencies as wildfly18 bom also has these jackson, netty dependencies.

Your thoughts

**Comment by alzimmermsft** [comment]
@SureshRangachariDXC, yes, it is possible that the dependencies being pulled in by WildFly are causing a runtime conflict. If the runtime environment loads Jackson 2.9 instead of the JAR's dependency on Jackson 2.12 the exception you mentioned will be thrown. This is especially tricky as the `XmlMapper` extends `ObjectMapper` from another Jackson package, so the reflective calls could find the expected methods on the classpath but the super class could be from another Jackson version which leads to an exception being thrown.

When you attempted to create a shaded JAR for the standalone application did the same exception occur?

**Comment by SureshRangachariDXC** [comment]
@alzimmermsft - We are able to print the blob as a stream in eclipse console when we run from a stand alone application 
But when this blobservice sdks are used in the wildfly project controllers classes and during the web page execution we end up getting the jackson related exception. So what do you think can be done here based on your experience.

**Comment by SureshRangachariDXC** [comment]
@alzimmermsft - we have tried to use wildfly 24 version in pom.xml because that was having dependency of jackson 2.12.3 versions but still we get the error. Can some one assist by a working session and this is impacting the deliverable.

**Comment by SureshRangachariDXC** [comment]
@anuchandy @alzimmermsft       

There is some dependency conflict. i See there is a similar issue at this post and which was solved anuchandy on below link

https://github.com/Azure/azure-sdk-for-java/issues/20094

---------------------------------------------------------------------------------------------------- I have explicitly provided jackson 2.12.3 on my pom with azure-storage-blob 12.12.0 version
Exception Details:
  Location:
    com/fasterxml/jackson/databind/cfg/MapperBuilder.streamFactory()Lcom/fasterxml/jackson/core/TokenStreamFactory; @7: areturn
  Reason:
    Type 'com/fasterxml/jackson/core/JsonFactory' (current frame, stack[0]) is not assignable to 'com/fasterxml/jackson/core/TokenStreamFactory' (from method signature)
  Current Frame:
    bci: @7
    flags: { }
    locals: { 'com/fasterxml/jackson/databind/cfg/MapperBuilder' }
    stack: { 'com/fasterxml/jackson/core/JsonFactory' }
  Bytecode:
    0000000: 2ab4 0002 b600 08b0                    

	at javax.faces.api@3.0.0.Final//javax.faces.webapp.FacesServlet.executeLifecyle(FacesServlet.java:725)

**Comment by SureshRangachariDXC** [comment]
[stacktrace.txt](https://github.com/Azure/azure-sdk-for-java/files/6840107/stacktrace.txt)

**Comment by SureshRangachariDXC** [comment]
@alzimmermsft @anuchandy - The issue has been fixed. Issue was with the transitive dependencies that was brought in by the application server run time modules (in our case it is Wildfly 18). On that note we have had to exclude the jackson core, annotations, databind, jackson2 provider...etc from our jboss deployment structure so that these modules that are part of run time wont load firstly on the class path conflicting azure sdk's dependency of jackson. We didnt do anything on shading as we were using wildfly BOM than individual dependencies.

Closing this thread.

Thank You.
